frostsnap_core/src/message.rs

```diff
@@ -52,16 +53,17 @@ pub enum CoordinatorToDeviceMessage {
 pub struct SignRequest {
     // TODO: explain these `usize` and create a nicely documented struct which explains the
     // mechanism
-    pub nonces: BTreeMap<DeviceId, (Vec<Nonce>, usize, usize)>,
+    pub nonces: BTreeMap<Scalar<Public, NonZero>, (Vec<Nonce>, usize, usize)>,
     pub sign_task: SignTask,
+    pub targets: BTreeSet<DeviceId>,
```

Comment:
I don't think it makes sense to put the target destinations here. The devices simply don't need to know this so why send them over the wire here. The devices figure out if it's meant for them in the `frostsnap_comms::CoordinatorSendMessage.target_destinations` field.

I do see the problem you're facing. We need to make who the targets are explicit because of this change. The code that generates the message needs to set the targets. I think we can do:

```rust
#[derive(Clone, Debug)]
pub enum CoordinatorSend {
    ToDevice { message: CoordinatorToDeviceMessage, destinations: BTreeSet<DeviceId>},
    ToUser(CoordinatorToUserMessage),
    ToStorage(CoordinatorToStorageMessage),
}
```

so that `start_sign` or whatever method that returns a CoordinatorSend can explicitly set the destinations.